sumologic/data_source_sumologic_role.go

```diff
@@ -0,0 +1,81 @@
+package sumologic
+
+import (
+	"errors"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceSumologicRole() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceSumologicRoleRead,
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"filter_predicate": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"capabilities": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func dataSourceSumologicRoleRead(d *schema.ResourceData, meta interface{}) error {
+	c := meta.(*Client)
+
+	var role *Role
+	var err error
+	if rid, ok := d.GetOk("id"); ok {
+		id := rid.(string)
+		role, err = c.GetRole(id)
+		if err != nil {
+			return fmt.Errorf("role with id %d not found: %v", id, err)
```

Comment:
there is a build error - id is not a decimal

Reply:
Fixed